core/src/main/java/cocktailofferer/service/image/ImageMetadata.java

```diff
@@ -0,0 +1,28 @@
+package cocktailofferer.service.image;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.*;
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Getter
+@EqualsAndHashCode(of = "id")
+public class ImageMetadata {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @NotNull
+    @Size(min = ImageMetadataConstraints.Key.MIN_SIZE, max = ImageMetadataConstraints.Key.MAX_SIZE)
+    @Column(unique = true)
+    private String imageKey;
```

Comment:
It's not obvious what a key is

Reply:
Suggest changes)

Reply:
It's not obvious, therefore I don't know what a key is needed for :shrug:

Reply:
Unique key of image in file system or object storage

Reply:
yes, object storage has a concept of key, but file system doesn't

Reply:
So, let's name it `s3Key`

Reply:
And when you encountered the situation when it was impossible to use `key` as table's attribute name (as it's a reserved keyword), you shouldn't have changed the class's field name because `imageKey` duplicates a piece of information (as class name contains the word `image`). You'd better change the table's attribute name to `key_`.